tests/e2e/topology_site_down_cases.go

```diff
@@ -0,0 +1,1416 @@
+		TESTCASE-1
+		Bring down partial sites zone1 and zone2 hosts (1 esx in zone1 + 1 esx in zone2)
+		Steps//
+		1. Create SC with waitForFirstConsumer
+		2. Create 3 statefulsets each with replica count 7
+		3. Wait for all the workload pods to reach running state
+		4. Verify that all the POD's spread across zones
+		5. Verify that the PV's have node affinity details
+		6. In zone1 bring down 1 ESX host and in zone2 bring down 1 ESX
+		7. Wait for some time and verify that all the node VM's are bought up on the ESX which was
+		in ON state in the respective clusters
+		8. Verify that All the workload POD's are up and running
+		9. Verify that the Node affinity details on the PV holds the zone and region  details
+		10. scale up any one stateful set to 10 replica's
+		11. Wait for some time and make sure all 10 replica's are in running state
+		12. Scale down the stateful set to 5 replicas
+		13. wait for some time and verify that stateful set used in step 12 has 5 replicas running
+		14. Bring up all the ESX hosts
+		15. Wait for testbed to be back to normal
+		16. Verify that all the POD's and sts are up and running
+		17. Verify All the PV's has proper node affinity details.
+		18. Verify pod and  pvc volume metadata in CNS.
+		19. Make sure K8s cluster is healthy
+		20. Delete above created STS, PVC's and SC
+	*/
+
+	ginkgo.It("Volume provisioning when partial sites zone1 and zone2 hosts are down", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		sts_count = 3
+		statefulSetReplicaCount = 7
+		noOfHostToBringDown = 1
+		var ssPods *v1.PodList
+
+		// Get allowed topologies for Storage Class
+		allowedTopologyForSC := getTopologySelector(topologyAffinityDetails, topologyCategories,
+			topologyLength)
+
+		// Create SC with WFC BindingMode
+		ginkgo.By("Creating Storage Class with WFC Binding Mode and allowed topolgies of 5 levels")
+		storageclass, err := createStorageClass(client, nil, allowedTopologyForSC, "",
+			bindingMode, false, "nginx-sc")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		// Creating StatefulSet service
+		ginkgo.By("Creating service")
+		service := CreateService(namespace, client)
+		defer func() {
+			deleteService(namespace, client, service)
+		}()
+
+		// Create multiple StatefulSets Specs in parallel
+		ginkgo.By("Creating multiple StatefulSets specs in parallel")
+		statefulSets := createParallelStatefulSetSpec(namespace, sts_count, statefulSetReplicaCount)
+
+		// Trigger multiple StatefulSets creation in parallel.
+		ginkgo.By("Trigger multiple StatefulSets creation in parallel")
+		var wg sync.WaitGroup
+		wg.Add(3)
+		for i := 0; i < len(statefulSets); i++ {
+			go createParallelStatefulSets(client, namespace, statefulSets[i],
+				statefulSetReplicaCount, &wg)
+		}
+		wg.Wait()
```

Comment:
Add a defer function

		defer func() {
			ginkgo.By(fmt.Sprintf("Deleting all statefulsets in namespace: %v", namespace))
			fss.DeleteAllStatefulSets(client, namespace)
		}()